lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2896,6 +2896,10 @@ void SplitEmitter::emitModule(EmittedModule &mod) {
 //===----------------------------------------------------------------------===//
 
 LogicalResult circt::exportVerilog(ModuleOp module, llvm::raw_ostream &os) {
+  // Since this code creates wire operations, we must make sure that the SV
+  // dialect is loaded.  The dialect will not have been loaded if the input IR
+  // has no SV operations.
+  module->getContext()->loadDialect<sv::SVDialect>();
```

Comment:
I assume there's no cost to loading this unconditionally if this was already loaded somewhere else? (Is there any reason to gate this check?)

Reply:
Its a pretty common pattern to call this function unconditionally. Looking at the code, it checks if the dialect has already been loaded by calling `getOrLoadDialect()` under the hood.

Reply:
Does this also need to register the dialect? I haven't dug into what getOrLoadDialect does recently, but I wonder what happens if it hasn't been registered yet. Not sure if this is actually a situation we care about, or where that registration happen right now for ExportVerilog.

Reply:
I think that you don't need a register a dialect to load it. The global registry is used to map a dialect string `"rtl"` to a dialect class `RTLDialect`.  This is used by the parser to lazily load dialects it encounters in the IR.  It looks like the dialects are inserted into the registry by the `registerToVerilogTranslation()`, which let's them be loaded lazily by `circt-translate` when *parsing*. In `firtool`, all the dialects we care about are loaded up-front without using the registry.  There is a bit more information here: https://mlir.llvm.org/getting_started/Faq/#registered-loaded-dependent-whats-up-with-dialects-management.

Reply:
Since its loaded up-front by firtool, maybe I should only load this before calling export-verilog in the translation callback?  I think this would be a bit better, and I can update the exportVerilog doc to say that it requires loading the SV dialect before calling.

Reply:
Got it, sorry I missed that the SV dialect (and others) are already registered by this translation... Some fresh scars here. Looks great.